src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -390,10 +390,14 @@ protected sealed override void LazyAsyncMethodChecks(CancellationToken cancellat
 
             if (!this.IsAsync)
             {
-                state.NotePartComplete(CompletionPart.StartAsyncMethodChecks);
-                if (state.NotePartComplete(CompletionPart.FinishAsyncMethodChecks) && IsPartialDefinition)
+                if (state.NotePartComplete(CompletionPart.StartAsyncMethodChecks))
                 {
-                    DeclaringCompilation.SymbolDeclaredEvent(this);
+                    if (IsPartialDefinition) DeclaringCompilation.SymbolDeclaredEvent(this);
```

Comment:
The events for methods normally contain a semantic model, and are produced in the method compilation pipeline. But partial methods have two different declarations in source.

This code is likely to change soon due to a separate issue with duplicate events for partial methods (both C# and VB). This code change helps keep existing tests working, but I don't think it would make sense to try to rationalize and document the current strategy which doesn't even work correctly.